packages/backend/src/tools/types.ts

```diff
@@ -32,3 +32,17 @@ export function stringAs<T>(Brand: (s: string) => T) {
     })
     .transform(Brand)
 }
+
+export function numberAs<T>(Brand: (n: number) => T) {
+  return z
+    .number()
+    .refine((s) => {
+      try {
+        Brand(s)
+        return true
+      } catch {
+        return false
+      }
+    })
+    .transform(Brand)
+}
```

Comment:
This could be abstracted away the following way:
```

function branded<T extends z.ZodTypeAny, B>(
  schema: T,
  Brand: (t: z.TypeOf<T>) => B,
) {
  return schema
    .refine((s: T) => {
      try {
        Brand(s)
        return true
      } catch {
        return false
      }
    })
    .transform(Brand)
}

export function stringAs<T>(Brand: (t: string) => T) {
  return branded(z.string(), Brand)
}

export function numberAs<T>(Brand: (t: number) => T) {
  return branded(z.number(), Brand)
}
```